app/controllers/check_errors_controller.rb

```diff
@@ -8,9 +8,21 @@ def confirm_clear_answer
     return render_not_found unless @log
 
     @related_question_ids = params[@log.model_name.param_key].keys.reject { |id| id == "page_id" }
-    question_id = @related_question_ids.find { |id| !params[id].nil? }
-    @question = @log.form.get_question(question_id, @log)
     @page = @log.form.get_page(params[@log.model_name.param_key]["page_id"])
+
+    if params["clear_all"]
+      @questions_to_clear = @related_question_ids.map { |id|
+        question = @log.form.get_question(id, @log)
+        next if question.subsection.id == "setup"
+
+        question.page.questions.map(&:id)
```

Comment:
Should this be just question? I know we don't put more than one on a page very often (and only if they are linked), but I'd expect others on the same page to be related in their own right if they needed to be cleared. (Maybe that expectation is wrong though)

Reply:
So there kinda is a reason I'm doing this. I think some of our pages have additional validations checking that when there are 2 questions on the page, all of the questions have to be answered. So when we choose to clear a single question from a page with 2 questions, we get a validation error. For example, if we have uprn known - yes and a uprn value, when we choose to clear uprn to correct a composite error, the log would go into an invalid state (another example of this where the questions aren’t conditional is income frequency and earnings for lettings)